lib/bolt/plugin/terraform.rb

```diff
@@ -49,11 +71,32 @@ def inventory_targets(opts)
       end
 
       def load_statefile(opts)
+        statefile = if opts['backend'] == 'remote'
```

Comment:
Do we need the `load_local_statefile` path at all or can we just always use `terraform state pull`, which should load from the local state if that's what's in use?

Reply:
Confirmed that terraform pull appears to work even when a remote backend is not used with both 0.11 and 0.12. Could there be a case where the user specifies a `statefile` that the CLI would not be configured to "pull" from?

Reply:
As far as I can tell, `terraform state pull` will only read a local file when it's named `terraform.tfstate`. So if a user had a state file with a different name then the plugin couldn't read from it. Not sure if this is a concern considering terraform itself won't read from these files.

Reply:
I'm also investigating another bug. Basically if you have a statefile that is version 3 (presumably generated by terrafomr < 0.12) and execute `terraform state pull` with terraform > 0.12 you get a `version: 4` state file that does not actually work with our extract_resources method. So unless I find an easy fix for that case, this could be another reason to still support parsing a statefile without using the CLI.